kernel/signal.c

```diff
@@ -899,8 +899,15 @@ long myst_signal_clone(
 
 void myst_handle_host_signal(siginfo_t* siginfo, mcontext_t* mcontext)
 {
+    assert(siginfo != NULL);
+    assert(mcontext != NULL);
+
     _handle_one_signal(siginfo->si_signo, siginfo, mcontext);
 
+    /* resume the execution based on mcontext for the unlikely case
+     * where the _handle_one_signal returns (e.g., early return) */
+    myst_sigreturn(mcontext);
```

Comment:
Is it OK to do this or are we masking a genuine issue by choosing to resume execution?